### ISSUE TITLE
hoistTry rule known issues

### ISSUE THREAD

**Issue by RomanTysiachnik** [nicklockwood/SwiftFormat#1462]
Hey @nicklockwood ,

I've recently attempted to apply `hoistTry` rule from swiftformat `0.51.11`, and here're some examples where it has corrupted my code:

1. Optional closure
```diff
                do {
-                    onCallback?(try decoder.decode(Message.self, from: data))
+                    onCallback?try (decoder.decode(Message.self, from: data))
                } catch {
                    print("\(error)")
                }
```

2. Generics
```diff
    func map<T>(_ transform: (Element) throws -> T) rethrows -> Tree<T> {
        return Tree<T>(
-            storage: Tree<T>.Storage(
-                elements: try storage.elements.map(transform)
+            storage: Tree<Ttry>.Storage(
+                elements: storage.elements.map(transform)
            )
        )
    }
```

3. Array

``` diff
-        if [.first, .second].contains(try element.unwrap()) {
+        if [.first, .secondtry].contains(element.unwrap()) {

```

Please let me know if you need more context to reproduce these.

**Comment by nicklockwood** [comment]
@RomanTysiachnik fixed in 0.51.12

**Comment by RomanTysiachnik** [comment]
Hey @nicklockwood ,

Thanks for fixing previous issues. Getting back to you with a new one :D 

```diff
    public func map<T>(_ transform: (Element) throws -> T) rethrows -> Tree<T> {
-       return Tree<T>(
+       return Tree<T>try (
            storage: Tree<T>.Storage(
-                elements: try storage.elements.map(transform),
+                elements: storage.elements.map(transform),
            )
        )
```

**swiftformat@0.51.12**
